core/src/avm2/class.rs

```diff
@@ -108,6 +110,44 @@ fn do_trait_lookup<'gc>(
     Ok(())
 }
 
+/// Find traits in a list of traits matching a slot ID.
+///
+/// This function also enforces final/override bits on the traits, and will
+/// raise `VerifyError`s as needed.
+///
+/// TODO: This is an O(n^2) algorithm, it sucks.
+fn do_trait_lookup_by_slot<'gc>(
+    id: u32,
+    known_traits: &mut Vec<Trait<'gc>>,
+    all_traits: &[Trait<'gc>],
+) -> Result<(), Error> {
+    for trait_entry in all_traits {
+        let trait_id = match trait_entry.kind() {
+            TraitKind::Slot { slot_id, .. } => slot_id,
+            TraitKind::Const { slot_id, .. } => slot_id,
+            TraitKind::Class { slot_id, .. } => slot_id,
+            TraitKind::Function { slot_id, .. } => slot_id,
+            _ => continue,
+        };
+
+        if id == *trait_id {
+            for known_trait in known_traits.iter() {
+                if known_trait.is_final() {
+                    return Err("Attempting to override a final definition".into());
+                }
+
+                if !trait_entry.is_override() {
+                    return Err("Definition override is not marked as override".into());
+                }
```

Comment:
Is this the right place for this? Seems like class/method overriding checks would occur when the ABC is initialized and the traits are loaded, and/or in `newclass`.

Methods don't occupy slots, so `override` doesn't apply here? Final would be based on `class.super_class` for classes and `method_info` for methods, the slot ID wouldn't affect this.